src/views/Game/GameDock.tsx

```diff
@@ -311,6 +312,18 @@ export function GameDock({
             ? engine.playerToMove()
             : engine.playerNotToMove();
 
+    const showLibraryModal = () => {
+        // Pull user data prior to opening modal preventing component from rendering prior to obtaining needed data
```

Comment:
We need to open the modal and show a Loading indicator, you can probably just drop in `<Loading />` while you wait for the data, that way people know their action has been registered and they just need to be patient while we wait for the internet.